modules/features/cu_article/cu_related_articles/cu_related_articles.module

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * Implements hook_block_info().
+ */
+function cu_related_articles_block_info() {
+
+  $blocks['related_articles_block_grid'] = array(
+    'info' => t('Related Articles Grid'),
+  );
+  $blocks['related_articles_block_column'] = array(
+    'info' => t('Related Articles Column'),
+  );
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function cu_related_articles_block_view($delta = '') {
+  switch ($delta) {
+
+    case 'related_articles_block_grid':
+      $block = cu_related_articles_block_output('grid');
+      break;
+    case 'related_articles_block_column':
+      $block = cu_related_articles_block_output('sidebar');
+      break;
+  }
+  return $block;
+}
+
+function cu_related_articles_block_output($view_mode) {
+  $related_articles = array();
+  // Get current node
+  if (($node = menu_get_object()) && $node->type =='article') {
+    $current_nid = $node->nid;
+    // Get categories from current article
+    $category_terms = $node->field_article_categories[LANGUAGE_NONE];
```

Comment:
This line throws an error if no categories are present. The !empty() check could assign the variable there, and I'd think the error would go away...or it could also need an isset().